updater/internal/logging/logging.go

```diff
@@ -0,0 +1,64 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logging
+
+import (
+	"fmt"
+
+	"github.com/observiq/observiq-otel-collector/updater/internal/path"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"gopkg.in/natefinch/lumberjack.v2"
+)
+
+// NewLogger returns a new logger, that logs to the log directory relative to installDir,
+// with the provided log level.
+func NewLogger(installDir string, level zapcore.Level) (*zap.Logger, error) {
+	prodConf := zap.NewProductionConfig()
+
+	prodLogger, err := prodConf.Build(zap.WrapCore(func(c zapcore.Core) zapcore.Core {
+		return core(installDir, level)
+	}))
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to make logger: %w", err)
+	}
+
+	return prodLogger, nil
+}
+
+func core(installDir string, level zapcore.Level) zapcore.Core {
+	logger := &lumberjack.Logger{
+		Filename:   path.LogFile(installDir),
+		MaxSize:    10,
+		MaxBackups: 3,
+	}
```

Comment:
We don't need log rotating in here can we remove this and reduce dependencies?

Reply:
Removed this, but won't the log file for the updater just end up growing without bound? I guess it would take quite a few updates to be significant.

Reply:
Yes, the updater should probably clean up and previous logs when it starts up. So updates over time don't continuously append on that file.

Reply:
Makes sense, the updater now deletes the previous log file when it creates the logger.